pss/outbox/outbox_test.go

```diff
@@ -115,14 +114,20 @@ func TestOutboxWorkers(t *testing.T) {
 	var maxParallel int32
 
 	var wg sync.WaitGroup
+	var mtx sync.Mutex
 	mockForwardFunction := func(msg *message.Message) error {
-		atomic.AddInt32(&parallel, 1)
+		mtx.Lock()
```

Comment:
why the change? it is more readable with `atomic` IMO

Reply:
There was an `atomic.AddInt32(&parallel, 1)` followed by a `parallel++` and an if check without a lock. It might not be super important it tests, but I guess that is not thread safe?

Reply:
cool. nice catch